agcaballero/walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -35,144 +46,130 @@
-      mandatory = true;
-    }
-    
-    // are there any optional employees?
-    boolean optional = false; 
-    if(optionalAttendees.size() > 0) {
-      optional = true;
-    }
+    // Holds whether the mandatory attendees can attend & the number of optional attendees 
+    // that can attend. If minuteStatus[number] is equal to MANDATORY_UNAVAILABLE, then mandatory
+    // attendees can't attend at that number. If it's equal to 0, then all attendees can attend.
+    // If it's equal to a negative int, then the abs value of that negative number is the number 
+    // of optional attendees who can't attend. The +1 is to account for last minute in day
+    int[] minuteStatus = new int[MINUTES_IN_DAY + 1];
 
-    // minutes of day is + 1 to account for first and last minute of day
-    int[] minutes = new int[MINUTES_IN_DAY + 1];
-    // 0 - all attendees (mandatory & optional) can attend 
-    // 1 - mandatory attendees can attend (but not all optional)
-    // 2 - not all mandatory attendees can attend (even if all optional can)
+    // keep track of min status reached 
+    int minStatus = 0;
 
     for(Event event: events) {
       int status = 0;
-      
-      if (attendeeOverlap(event.getAttendees(), attendees)) {
-        status = 2;
-      }
-      else if (attendeeOverlap(event.getAttendees(), optionalAttendees)) {
-        status = 1;
+      int mandatoryOverlap = amountOfOverlap(event.getAttendees(), attendees);
+
+      if(mandatoryOverlap > 0) {
+        status = MANDATORY_UNAVAILABLE;
+      } 
+      else {
+        int optionalOverlap = amountOfOverlap(event.getAttendees(), optionalAttendees);
+        
+        if(optionalOverlap > 0) {
+          status = optionalOverlap;
+        }
       }
-
-      if (status != 0) {
-
+       
+      // 0 is default so does not need to be set
+      if(status != 0) {
         TimeRange range = event.getWhen();
 
-        for(int i = range.start(); i < range.end(); i++)
-          // make sure 2s (which are higher priority than 1s) are not overwritten
-          if(minutes[i] != 2)
-            minutes[i] = status;
+        for(int i = range.start(); i < range.end(); i++) {
+          // don't change it if it's already set to being unavailable
+          if(minuteStatus[i] != MANDATORY_UNAVAILABLE) {
+            if(status == MANDATORY_UNAVAILABLE) {
+              minuteStatus[i] = MANDATORY_UNAVAILABLE;
+            }
+            else {
+              // if it's available but some attendees can't make it
+              // then just subtract the ones who can't make it from 
+              // the status (so the number of optional attendees who can't make
+              // it will compound)
+              minuteStatus[i] -=  status;
+
+              // if a new minimum status is found, update the min status champion
+              if (minuteStatus[i] < minStatus) {
+                minStatus = minuteStatus[i];
+              }
+            }
+          }
+        }
```

Comment:
My guess is that the purpose of this block of code is to populate the `minuteStatus` array, but I'm pretty confused what's going on. I think maybe clearer variable naming would help. In general favor longer names that provide enough description. May take longer to type, but will speed up review times.

Reply:
I think the use of the Availability class and implementation of the other edits should help. Please let me know if it needs more clarification.

Reply:
Done